src/avatar.jsx

```diff
@@ -156,7 +154,7 @@ const Avatar = React.createClass({
           className={className}
         >
           {icon && React.cloneElement(icon, {
-            color: color,
+            color: color || avatar.color,
             style: Object.assign(styles.icon, icon.props.style),
```

Comment:
Oh that's a good point! That was probably the original intention, to abstract that detail. Mmm...